### DISCUSSION TITLE
How to use $ signs?

### DISCUSSION THREAD

**Discussion by maral** [noam-honig/radweb#197] [Q&A] [answered]
Hi, I wonder how can I use the $ signs, like in this code: `await task.$.category!.load()` VS Code does not recognize them, is there some extra configuration step? I haven't found it mentioned anywhere...

**Comment by noam-honig** [comment, marked as answer]
Hi Marek,

There are two styles of writing when using remult.

One is referred to as  - ActiveRecord, in that case you update the existing
state of an object - when using this style, you can inherit from
`EntityBase` and then you'll have the $ sign.

For example:
task.$.title.displayValue


The other style, uses Immutable objects - in that case, the $ sign is less
applicable, but you can use the fields member of the repository to get a
similar result.

For example
const repo = remult.repo(Task)
repo.fields.title.displayValue(task)


Does that help?

**Reply by maral** [reply]
Thanks for a quick reply! The trick with repo.fields looks good, but I can't use it for lazy loading that way. But glad I asked, I will try to find how to use it exactly.

I really like remult, but I have some things I just can't find in the documentation (so I am looking around the repos using remult and trying to copy what they are doing - but some I cannot find). Can you please answer me these other questions I could not find answer to?

1. Is there some simple way to log all SQL queries, ideally to the console or file?
2. Is there some recommended way to use sqlExpression in an entity field? I don't really know how to use it... How can I reference other columns, can I use Knex query builder somehow, etc.?
3. Let's say I am loading 100 records from one table. Each record has a FK to another table, which has another FK to another table. I would like to load values from the row -> FK -> FK -> column to a field. And ideally, I would want to load those columns from the referenced tables in one query, not in 100 queries for each row. Is that possible?
4. Would you recommend some repo that has remult implemented well, so I can learn how they do things?

**Comment by noam-honig** [comment]
Hi Marek, Happy to hear you are enjoying remult.

Regarding the Lazy loading using the repo.fields, let me know how you progress with it, and maybe there's room to improve the support with remult.

Regarding your questions:
1. Using the built-in Postgres adapter, you can activate sql logging using `SqlDatabase.LogToConsole = true`
  If you're using knex - then in the knex definitions you can say `debug:true` and you'll see all sqls in the console.
2. Regarding sqlExpressions - you can use them if you're using the postgres adapter, here's [an article](https://remult.dev/docs/techniques-regarding-one-to-many-relations.html#alternative-1) with several examples.
  If you're using knex, we don't yet support `sqlExpression` for knex, if you need it open a feature request and we'll add support for that. In the meantime, you can use a database view for these expressions, and set the `dbReadOnly` option for that field, that will cause the field to only be part of the select, but not in the update or insert process.
3. I often use `sqlExpression` for such scenarios, since it also allows me to filter and sort  based on these columns.
4. Sure, check out our [crm demo repo](https://www.github.com/remult/crm-demo), a crm application written with remult and react

You are also invited to join our discord channel at: https://discord.gg/GXHk7ZfuG5
